### ISSUE TITLE
Kismatic hangs at Validate kubernets Control Planel validation

### ISSUE THREAD

**Issue by zakir-hyder** [apprenda/kismatic#1249]
**Is this a BUG REPORT or FEATURE REQUEST?**:
BUG REPORT
**What happened**:
I'm trying to setup a cluster with 2 master, 3 etcd and 1 workers in proxmox.
There are two issues

1. Kube-apiserver is not runing
2. kubectl get cs returing
```
NAME                 STATUS      MESSAGE                                                                                                                                        ERROR
scheduler            Healthy     ok
controller-manager   Healthy     ok
etcd-2               Unhealthy   Get https://k8s-etcd03:2379/health: net/http: request canceled while waiting for connection (Client.Timeout exceeded while awaiting headers)
etcd-1               Unhealthy   Get https://k8s-etcd02:2379/health: net/http: request canceled while waiting for connection (Client.Timeout exceeded while awaiting headers)
etcd-0               Unhealthy   Get https://k8s-etcd01:2379/health: net/http: request canceled while waiting for connection (Client.Timeout exceeded while awaiting headers)
```
3. there is no files in etc/cni/net.d. log from master
```
cni.go:171] Unable to update cni config: No networks found in /etc/cni/net.d
kubelet.go:2125] Container runtime network not ready: NetworkReady=false reason:NetworkPluginNotReady message:docker: network plugin is not read
etc
```
**What you expected to happen**:
K8s cluster should be up and running

**How to reproduce it (as minimally and precisely as possible)**:
Setup vms in proxmox and run the kismatic.

**Anything else we need to know?**:

It is possible to connect to etcd cluster from master
```
curl --key etcd-client-key.pem --cacert ca.pem --cert etcd-client.pem https://k8s-etcd01:2379/v2/keys
{"action":"get","node":{"dir":true}}
```

**Environment**:
- Kismatic version (use `kismatic version`):
```
Kismatic:
  Version: 1.11.1
  Built: Wed May 30 20:47:57 UTC 2018
  Go Version: go1.9.4
```
- Plan file Gist (remove sensitive information if any): 
```
cluster:
  name: k8s

  # Kubernetes cluster version (supported minor version "v1.10.x").
  version: v1.10.2

  # Set to true if the nodes have the required packages installed.
  disable_package_installation: false

  # Set to true if you are performing a disconnected installation.
  disconnected_installation: false

  # Networking configuration of your cluster.
  networking:

    # Kubernetes will assign pods IPs in this range. Do not use a range that is
    # already in use on your local network!
    pod_cidr_block: 172.16.0.0/16

    # Kubernetes will assign services IPs in this range. Do not use a range
    # that is already in use by your local network or pod network!
    service_cidr_block: 172.20.0.0/16

    # Set to true if your nodes cannot resolve each others' names using DNS.
    update_hosts_files: true

    # Set the proxy server to use for HTTP connections.
    http_proxy: ""

    # Set the proxy server to use for HTTPs connections.
    https_proxy: ""

    # List of host names and/or IPs that shouldn't go through any proxy.
    # All nodes' 'host' and 'IPs' are always set.
    no_proxy: ""

  # Generated certs configuration.
  certificates:

    # Self-signed certificate expiration period in hours; default is 2 years.
    expiry: 17520h

    # CA certificate expiration period in hours; default is 2 years.
    ca_expiry: 17520h

  # SSH configuration for cluster nodes.
  ssh:

    # This user must be able to sudo without password.
    user: admin

    # Absolute path to the ssh private key we should use to manage nodes.
    ssh_key: /home/admin/.ssh/id_rsa
    ssh_port: 22

  # Override configuration of Kubernetes components.
  kube_apiserver:
    option_overrides: {}

  kube_controller_manager:
    option_overrides: {}

  kube_scheduler:
    option_overrides: {}

  kube_proxy:
    option_overrides: {}

  kubelet:
    option_overrides: {}

  # Kubernetes cloud provider integration.
  cloud_provider:

    # Options: 'aws','azure','cloudstack','fake','gce','mesos','openstack',
    # 'ovirt','photon','rackspace','vsphere'.
    # Leave empty for bare metal setups or other unsupported providers.
    provider: ""

    # Path to the config file, leave empty if provider does not require it.
    config: ""

# Docker daemon configuration of all cluster nodes.
docker:

  # Set to true if docker is already installed and configured.
  disable: false
  logs:
    driver: json-file
    opts:
      max-file: "1"
      max-size: 50m

  storage:

    # Leave empty to have docker automatically select the driver.
    driver: ""
    opts: {}

    # Used for setting up Device Mapper storage driver in direct-lvm mode.
    direct_lvm_block_device:

      # Absolute path to the block device that will be used for direct-lvm mode.
      # This device will be wiped and used exclusively by docker.
      path: ""
      thinpool_percent: "95"
      thinpool_metapercent: "1"
      thinpool_autoextend_threshold: "80"
      thinpool_autoextend_percent: "20"

# If you want to use an internal registry for the installation or upgrade, you
# must provide its information here. You must seed this registry before the
# installation or upgrade of your cluster. This registry must be accessible from
# all nodes on the cluster.
docker_registry:

  # IP or hostname and port for your registry.
  server: ""

  # Absolute path to the certificate authority that should be trusted when
  # connecting to your registry.
  CA: ""

  # Leave blank for unauthenticated access.
  username: ""

  # Leave blank for unauthenticated access.
  password: ""

# A set of files or directories to copy from the local machine to any of the nodes in the cluster.
additional_files: []

# Add-ons are additional components that KET installs on the cluster.
add_ons:
  cni:
    disable: false

    # Selecting 'custom' will result in a CNI ready cluster, however it is up to
    # you to configure a plugin after the install.
    # Options: 'calico','weave','contiv','custom'.
    provider: calico
    options:
      calico:

        # Options: 'overlay','routed'.
        mode: overlay

        # Options: 'warning','info','debug'.
        log_level: info

        # MTU for the workload interface, configures the CNI config.
        workload_mtu: 1500

        # MTU for the tunnel device used if IPIP is enabled.
        felix_input_mtu: 1440

        # Used to detect the IPv4 address of the host.
        ip_autodetection_method: first-found

      weave:

        # Used by Weave for network traffic encryption.
        # Should be reasonably strong, with at least 50 bits of entropy.
        password: ""

  dns:
    disable: false

    # Options: 'kubedns','coredns'.
    provider: kubedns
    options:
      replicas: 2

  heapster:
    disable: false
    options:
      heapster:
        replicas: 2

        # Specify kubernetes ServiceType. Defaults to 'ClusterIP'.
        # Options: 'ClusterIP','NodePort','LoadBalancer','ExternalName'.
        service_type: ClusterIP

        # Specify the sink to store heapster data. Defaults to an influxdb pod
        # running on the cluster.
        sink: influxdb:http://heapster-influxdb.kube-system.svc:8086

      influxdb:

        # Provide the name of the persistent volume claim that you will create
        # after installation. If not specified, the data will be stored in
        # ephemeral storage.
        pvc_name: ""

  # Metrics Server is a cluster-wide aggregator of resource usage data.
  metrics_server:
    disable: false

  dashboard:
    disable: false
    options:
      service_type: ClusterIP

  package_manager:
    disable: false

    # Options: 'helm'.
    provider: helm
    options:
      helm:
        namespace: kube-system

  # The rescheduler ensures that critical add-ons remain running on the cluster.
  rescheduler:
    disable: false

# Etcd nodes are the ones that run the etcd distributed key-value database.
etcd:
  expected_count: 3

  # Provide the hostname and IP of each node. If the node has an IP for internal
  # traffic, provide it in the internalip field. Otherwise, that field can be
  # left blank.
  nodes:
  - host: "k8s-etcd01"
    ip: "10.0.0.217"
    internalip: "10.0.0.217"

  - host: "k8s-etcd02"
    ip: "10.0.0.218"
    internalip: "10.0.0.218"

  - host: "k8s-etcd03"
    ip: "10.0.0.219"
    internalip: "10.0.0.219"

# Master nodes are the ones that run the Kubernetes control plane components.
master:
  expected_count: 2

  # If you have set up load balancing for master nodes, enter the FQDN name here.
  # Otherwise, use the IP address of a single master node.
  load_balanced_fqdn: "10.0.0.220"

  # If you have set up load balancing for master nodes, enter the short name here.
  # Otherwise, use the IP address of a single master node.
  load_balanced_short_name: "10.0.0.220"
  nodes:
  - host: "k8s-master01"
    ip: "10.0.0.220"
    internalip: "10.0.0.220"
    labels: {"master01"}
    taints: []

  - host: "k8s-master02"
    ip: "10.0.0.221"
    internalip: "10.0.0.221"
    labels: {"master02"}
    taints: []

# Worker nodes are the ones that will run your workloads on the cluster.
worker:
  expected_count: 1
  nodes:
  - host: "k8s-workers02"
    ip: "10.0.0.234"
    internalip: "10.0.0.234"
    labels: {"worker02"}
    taints: []

# Ingress nodes will run the ingress controllers.
ingress:
  expected_count: 0
  nodes: []

# Storage nodes will be used to create a distributed storage cluster that can
# be consumed by your workloads.
storage:
  expected_count: 0
  nodes: []
```
- Cloud provider or hardware configuration:

- OS (e.g. from /etc/os-release):
```
NAME="Ubuntu"
VERSION="16.04.4 LTS (Xenial Xerus)"
ID=ubuntu
ID_LIKE=debian
PRETTY_NAME="Ubuntu 16.04.4 LTS"
VERSION_ID="16.04"
HOME_URL="http://www.ubuntu.com/"
SUPPORT_URL="http://help.ubuntu.com/"
BUG_REPORT_URL="http://bugs.launchpad.net/ubuntu/"
VERSION_CODENAME=xenial
UBUNTU_CODENAME=xenial
```
- Kernel (e.g. `uname -a`):
```
Linux infra-exec 4.4.0-127-generic #153-Ubuntu SMP Sat May 19 10:58:46 UTC 2018 x86_64 x86_64 x86_64 GNU/Linux
```

**Comment by dkoshkin** [comment]
1. Check the logs for the api-server, `ssh` onto the master and use `docker logs` on the api-server to see why its failing.

3. The CNI is not yet setup until the api-server is running.

**Comment by zakir-hyder** [comment]
Thank you @dkoshkin for replying. I figured out the issue - it was network connectivity issue.

For reference
I am using Ubuntu 16.04 and it comes with Dnsmasq. This Dnsmasq was not reading from /etc/hosts. I also had a dnsservers record in my /etc/network/interfaces. I installed Dnsmasq and used following configurations
```
domain-needed
bogus-priv

domain=k8.com
expand-hosts
local=/k8.com/

listen-address=127.0.0.1
bind-interfaces

server=8.8.8.8
server=8.8.4.4
```

after that kismatic was able to install the k8s without any issue.
